base/main.tf

```diff
@@ -81,9 +81,13 @@ data "template_file" "helm_values" {
 }
 
 resource "null_resource" "helm_values_file" {
+  triggers {
+    content = "${data.template_file.helm_values.rendered}"
+  }
+
   provisioner "local-exec" {
     command = <<-EOC
-      tee helm_values.yaml <<EOF
+      tee ${path.cwd}/helm_values.yaml <<EOF
```

Comment:
no, the path doesn't trigger an update. This doesn't check the differences in your local `helm_values.yaml` file. It only checks the differences between the generated template and the one that's in the terraform state. That's what's the `triggers` for.